Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -351,11 +353,26 @@ private void OnDamageChanged(EntityUid uid, IgniteOnHeatDamageComponent componen
             if (args.DamageDelta == null)
                 return;
 
-            // Check if its' taken any heat damage, and give the value
-            if (args.DamageDelta.DamageDict.TryGetValue("Heat", out FixedPoint2 value))
+            // Check if we've taken any damage
+            if (args.DamageDelta.DamageDict.TryGetValue("Heat", out FixedPoint2 heatDamage))
             {
+                // Ignition can be prevented if the PreventedByHydration is true
+                // Reduce the user's hydration level by the damage times the mulitiplier
+                if (component.PreventedByHydration)
+                {
+                    if (TryComp<ThirstComponent>(uid, out var thirst))
+                    {
+                        // Thirst needs to be a negative value to go down
+                        EntityManager.System<ThirstSystem>().ModifyThirst(uid, thirst, heatDamage.Float() * -component.DehydrationMultiplier);
```

Comment:
You should add `[Dependency] private readonly ThirstSystem _thirst = default!;`
```suggestion
                        _thirst.ModifyThirst(uid, thirst, heatDamage.Float() * -component.DehydrationMultiplier);
```